fe_utils/mesh.py

```diff
@@ -56,6 +56,10 @@ def __init__(self, vertex_coords, cell_vertices):
             """The indices of the edges incident to each cell (only for 2D
             meshes)."""
 
+            # Here we define the derivatives matrix of the Lagrange P1 basis: We know that the nodal basis
+            # on the reference element is: x, y, 1-x-y.
+            self.lagrange_derivative = np.array([[-1, -1], [1, 0], [0,1]])
```

Comment:
Why are you hard-coding this for dimension 2 when you've already implemented lagrange elements and their tabulation?